### PR TITLE
Authorization update

### LINKED ISSUE #3076: There is no way to tell that authorization is required without a role or policy specified
@sungam3r I messed up a bit in the last PR.  The ASP.NET Core `[Authorize]` attribute when specified without a role or policy indicates that the user should be authenticated (not a guest).  We should have matching functionality in GraphQL.

This means that the `RequiresAuthorization` extension method should return `true` when a field is marked with `[GraphQLAuthorize]`, which it does not now.

I suggest:
- Add a new extension method to indicate that the user must be authenticated.  For example, called `RequireAuthentication` or `WithAuthorization()`.
- Modify the attribute to call the above method when no policies or roles are specified
- Modify the `RequiresAuthorization` extension method to return `true` when the above method is called.  (breaking change from yesterday; not breaking from 5.0.0)

I would like to do this promptly, while GraphQL 5 is still new.  Hopefully release 5.1.1 (or 5.2.0) tomorrow with this change.

### PR DESCRIPTION
**Pull request by Shane32**
Closes #3076

This also fixes the attribute so that it applies the metadata to query arguments or enum definitions also, if it is marked on those type of objects.

### CODE REVIEW

File: docs2/site/docs/migrations/migration5.md [hunk 1]
```diff
@@ -602,9 +602,10 @@ There are a number of other minor issues fixed; see these links for more details
 - https://github.com/graphql-dotnet/graphql-dotnet/issues/3002
 - https://github.com/graphql-dotnet/graphql-dotnet/pull/3004
 
-### 20. `AuthorizeWithRoles` added in GraphQL 5.1.0
+### 20. `Authorize` and `AuthorizeWithRoles` added in GraphQL 5.1.0
```
Comment (sungam3r) [inline docs2/site/docs/migrations/migration5.md:605]:
```suggestion
### 20. `Authorize` and `AuthorizeWithRoles` extension methods added in GraphQL 5.1.0
```

File: docs2/site/docs/migrations/migration5.md [hunk 2]
```diff
@@ -602,9 +602,10 @@ There are a number of other minor issues fixed; see these links for more details
...
 
 This allows for specifying roles rather than just policies that can be used to validate a request.
+`Authorize` can be used to specify that authentication is required, without specifying any specific roles or policies.
```
Comment (sungam3r) [inline docs2/site/docs/migrations/migration5.md:608]:
```suggestion
`Authorize` can be used to specify that only authentication is required, without specifying any specific roles or policies.
```

File: src/GraphQL/Authorization/AuthorizationExtensions.cs [hunk 3]
```diff
@@ -20,6 +20,12 @@ public static class AuthorizationExtensions
         /// </summary>
         public const string ROLE_KEY = "Authorization__Roles";
 
+        /// <summary>
+        /// Metadata key name for indicating that the user must be authorized to access the resource.
```
Comment (sungam3r) [inline src/GraphQL/Authorization/AuthorizationExtensions.cs:24]:
```suggestion
        /// Metadata key name for indicating that the user must be authorized (strictly speaking, authenticated) to access the resource.
```

**Review by sungam3r — APPROVED** [review]
Only cosmetic changes suggestions.

**Comment by Shane32** [comment]
5.1.1 or 5.2.0 ?

File: docs2/site/docs/migrations/migration5.md [hunk 4]
```diff
@@ -602,9 +602,10 @@ There are a number of other minor issues fixed; see these links for more details
 - https://github.com/graphql-dotnet/graphql-dotnet/issues/3002
 - https://github.com/graphql-dotnet/graphql-dotnet/pull/3004
 
-### 20. `AuthorizeWithRoles` added in GraphQL 5.1.0
+### 20. `Authorize` and `AuthorizeWithRoles` extension methods added in GraphQL 5.1.0
```
Comment (Shane32) [inline docs2/site/docs/migrations/migration5.md:605]:
```suggestion
### 20. `Authorize` and `AuthorizeWithRoles` extension methods added in GraphQL 5.1.1
```
?

Comment (sungam3r) [reply docs2/site/docs/migrations/migration5.md:605]: Write both.

**Comment by sungam3r** [comment]
5.1.1 since new APIs may be considered as a way to fix a bug

File: docs2/site/docs/migrations/migration5.md (hunk 4 above)
Comment (Shane32) [reply docs2/site/docs/migrations/migration5.md:605]:
```suggestion
### 20. `Authorize` and `AuthorizeWithRoles` extension methods added in GraphQL 5.1.0 and 5.1.1
```

Comment (Shane32) [reply docs2/site/docs/migrations/migration5.md:605]: Like this?

**Comment by Shane32** [comment]
> 5.1.1 since new APIs may be considered as a way to fix a bug

I agree

File: docs2/site/docs/migrations/migration5.md (hunk 4 above)
Comment (sungam3r) [reply docs2/site/docs/migrations/migration5.md:605]: yes

**Pull request merged**